src/setup/rgw.py

```diff
@@ -495,7 +454,97 @@ def _generate_ssl_cert(conf: MappedConf):
+        Log.info(f'Fetched motr_ha_endpoint from data pod. Endpoint: {motr_ha_endpoint}')
+
+        rgw_config_path = Rgw._get_rgw_config_path(conf)
+        Rgw._load_rgw_config(Rgw._rgw_conf_idx, f'ini://{rgw_config_path}')
+        Conf.set(Rgw._rgw_conf_idx, \
+            f'client.radosgw-admin>{RgwEndpoint.MOTR_HA_EP.value}', motr_ha_endpoint)
+        Conf.save(Rgw._rgw_conf_idx)
+
+        Log.info(f'Updated motr_ha_endpoint in config file {rgw_config_path}')
+
+        # admin user should be created only on one node.
+        # 1. While creating admin user, global lock created in consul kv store.
+        # (rgw_consul_index, cortx>rgw>volatile>rgw_lock, machine_id)
+        # 2. Before creating admin user.
+        #    a. Check for rgw_lock in consul kv store.
+        #    b. Create user only if lock value is None/machine-id.
+
+        rgw_lock = False
+        rgw_lock_key = f'component>{COMPONENT_NAME}>volatile>{COMPONENT_NAME}_lock'
+        rgw_consul_idx = f'{COMPONENT_NAME}_consul_idx'
+        # Get consul url from cortx config.
+        consul_url = Rgw._get_consul_url(conf)
+        # Check for rgw_lock in consul kv store.
+        Log.info('Checking for rgw lock in consul kv store.')
+        Conf.load(rgw_consul_idx, consul_url)
+
+        # if in case try-catch block code executed at the same time on all the nodes,
+        # then all nodes will try to update rgw lock-key in consul, after updating key
+        # it will wait for sometime(time.sleep(3)) and in next iteration all nodes will
+        # get lock value as node-id of node who has updated the lock key at last.
+        # and then only that node will perform the user creation operation.
+        while True:
+            try:
+                rgw_lock_val = Conf.get(rgw_consul_idx, rgw_lock_key)
+                Log.info(f'rgw_lock value - {rgw_lock_val}')
+                # TODO: Explore consul lock - https://www.consul.io/commands/lock
+                if rgw_lock_val is None:
+                    Log.info(f'Setting confstore value for key :{rgw_lock_key}'
+                             f' and value as :{Rgw._machine_id}')
+                    Rgw._load_rgw_config(rgw_consul_idx, consul_url)
+                    Conf.set(rgw_consul_idx, rgw_lock_key, Rgw._machine_id)
+                    Conf.save(rgw_consul_idx)
+                    Log.info('Updated confstore with latest value')
+                    time.sleep(3)
+                    continue
+                elif rgw_lock_val == Rgw._machine_id:
+                    Log.info('Found lock acquired successfully hence processing'
+                             ' with RGW admin user creation.')
+                    rgw_lock = True
+                    break
+                elif rgw_lock_val != Rgw._machine_id:
+                    Log.info(
+                        'Skipping rgw user creation, as rgw lock is already'
+                        f' acquired by {rgw_lock_val}')
```

Comment:
Skipping rgw user creation, as rgw lock is already'
                        f' acquired by other machine with id {rgw_lock_val}'

Reply:
done